EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/attributeInput/AttributeInput.java

```diff
@@ -0,0 +1,190 @@
+
+	AttributeInput(TypeTreeNode attribute) {
+		this.attribute = attribute;
+	}
+
+	public static AttributeInput attributeInputFactory(TypeTreeNode attribute) {
+		switch (attribute.getType()) {
+			case STRING:
+				return new StringAttributeInput(attribute);
+			case INTEGER:
+				return new IntegerAttributeInput(attribute);
+			case FLOAT:
+				return new FloatAttributeInput(attribute);
+			case DATE:
+				return new DateAttributeInput(attribute);
+			default:
+				attribute.setType(AttributeTypeEnum.STRING);
+				return new StringAttributeInput(attribute);
+		}
+	}
+
+	public TypeTreeNode getAttribute() {
+		return attribute;
+	}
+
+	public AttributeTypeEnum getAttributeType() {
+		return this.getAttribute().getType();
+	}
+
+	public String getAttributeName() {
+		return this.getAttribute().getName();
+	}
+
+	public List<ProbabilityDistributionEnum> getAvailableMethods() {
+		return availableMethods;
+	}
+
+	public ProbabilityDistributionEnum getSelectedMethod() {
+		return selectedMethod;
+	}
+
+	public void setSelectedMethod(ProbabilityDistributionEnum selectedMethod) {
+		this.selectedMethod = selectedMethod;
+	}
+
+	public String getInput() {
+		return input;
+	}
+
+	String getInputOrDefault() {
+		if(input != null && !input.isEmpty()) {
+			return input;
+		}
+		return this.getDefaultInput();
+	}
+
+	public void setInput(String input) {
+		this.input = input;
+	}
+
+	/**
+	 * Returns the chosen value from user input. If no value was chosen up to now, this will be done first.
+	 *
+	 * @return a serializable object containing a single value chosen from user input
+	 */
+	public Serializable getCalculatedValue() {
+		if (this.getValue() == null) {
+			this.calculateRandomValue();
+		}
+		return this.getValue();
+	}
+
+	/**
+	 * Chooses a value from the user input and saves it in the value-instance variable.
+	 *
+	 */
+	public abstract void calculateRandomValue();
+
+	/**
+	 * Abstract getter for the value instance variable defined in subclasses.
+	 * @return a serializable object
+	 */
+	abstract Serializable getValue();
+
+	/**
+	 * Returns the chosen value as string instead of serializable object.
+	 * It will execute the calculation of a value in case this wasn't done until now.
+	 *
+	 * @return a string
+	 */
+	public String getValueAsString() {
+		return String.valueOf(this.getCalculatedValue());
+	}
+
+	String getDefaultInput() { return "UNDEFINED"; }
+
+	/**
+	 * Checks whether the user input is contained in a given range.
+	 * Might get overridden in subclass.
+	 *
+	 * @param range to be searched in
+	 * @return bool if string is in range
+	 */
+	public boolean isInRange(String range) {
+		String[] possibleValues = range.split(";");
+		for (String possibleValue : possibleValues) {
+			if (possibleValue.equals(this.getValueAsString())) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	@Override
+	public String toString() {
+		StringBuilder stringify = new StringBuilder(System.getProperty("line.separator"));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "line.separator" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1192)